### ISSUE TITLE
Add-on GUI and using other XBMC / Kodi Skins - Problems

### ISSUE THREAD

**Issue by cw-kid** [gderosa/xbmc-vera#9]
Hi

The Vera add-on for XBMC / Kodi generally works well with the default Kodi skin called Confluence. 

However I have tried to use the Vera add-on with other Kodi skins and the results are not good. 

For example here is the Vera add-on in Kodi version 14 (Helix) when using the default Kodi skin Confluence. As you can see in this screen shot the GUI looks OK and everything is visible as it should be.

![vera-addon-confluence](https://cloud.githubusercontent.com/assets/9383216/6843674/07e6a970-d39d-11e4-9c71-35f4656052bf.png)

However if I change the Kodi skin to the skin I am currently using Xperience1080: http://forum.kodi.tv/showthread.php?tid=210425

The Vera add-on then looks like this:

![vera-addon-xperience1080](https://cloud.githubusercontent.com/assets/9383216/6843720/625c6ae8-d39d-11e4-81f3-a0e907db8ef2.png)

The GUI now looks incorrect and the button text and button boxes are all missing. When you highlight a button by moving around the GUI the text for that button is only then displayed.

**Comment by cw-kid** [comment]
Threads about this problem on the Vera forums:

http://forum.micasaverde.com/index.php/topic,28007.msg216296.html#msg216296
http://forum.micasaverde.com/index.php/topic,28007.msg216712.html#msg216712

**Comment by gderosa** [comment]
@cw-kid, I'm not surprised. I recall that I had to do lots of hardcoding which was not very portable across skins...

All the presentation layer is in dire need of a substantial refactoring. This will take _some time_...

In the meantime: thanks to you and all the community for the continuing and growing interest, and my apologies for not being able to do much in the last few months/years.

Also, thanks in advance to whoever can donate testing hardware (e.g. a Vera controller at least) including a remote access (send an email or pm for the details). 

All the best.

Guido

**Comment by cw-kid** [comment]
Guido 

I could give you remote access to my Vera unit if that helps? Please email me. 

I have also asked the Z-wave distributor if they have an old Vera controller unit they can donate to this project? Fingers crossed.

**Comment by gderosa** [comment]
Thanks a lot @cw-kid!

Old... but not too old? I mean, i guess many versions of Vera have existed over the years, and in case of incompatible APIs I guess we want to support the newest ones? This will require some investigation...

**Comment by cw-kid** [comment]
I assume it's a Vera 3 or Veralite. They have agreed to donate a unit to this project. I didn't ask which model it was.

**Comment by cw-kid** [comment]
Old was the wrong word, I requested a spare or returned unit, please confirm your shipping address and I will ask which model they are sending? You never know it might be a new VeraEdge!

**Comment by cw-kid** [comment]
I am still using Vera UI5 on my Veralite unit. As it's stable and all the 3rd party mios add-ons I am using work with it. The current version of the Vera OS is UI7. VeraEdge ships with UI7. However Vera 3 and Veralite can be updated to UI7. There are still a lot of Vera users still using UI5 however for the same reasons as me. I have had other users on the Vera forum confirm that your XBMC add-on that I modified works with UI7. And I have it working with UI5.

**Comment by gderosa** [comment]
So, right now, UI5 is still the most popular and widely supported, if I understand it correctly?

Let's go for it so. (Maybe tthe JSON API hasn't changed at all with UI6 and UI7?)

I think I have at least access to a remote UI5 already...  But I'll let you know more shortly, thanks very much for now.

**Comment by cw-kid** [comment]
Correct, Yes I believe most people still using UI5 rather than UI7. 
And I don't think the JSON API has changed as people claim to be using your add-on ok with UI7 Veras.

**Comment by cw-kid** [comment]
This document is about writing plugins for Vera itself, but it does refer to a new JavaScript API in UI7 does this affect your XBMC Add-on? 

http://wiki.mios.com/index.php/JavaScript_API

This document talks about creating remote control apps and does not mention UI7 at all

http://wiki.micasaverde.com/index.php/UI_Notes

**Comment by gderosa** [comment]
> This document is about writing plugins for Vera itself, but it does refer to a new JavaScript API in UI7 does this affect your XBMC Add-on? 
> 
> http://wiki.mios.com/index.php/JavaScript_API

I don't think so. As you say, it's for plugins running within Vera itself, not for talking to Vera from remote.

> This document talks about creating remote control apps and does not mention UI7 at all
> 
> http://wiki.micasaverde.com/index.php/UI_Notes

Yep. This seems to confirm the remote API has beeen kept stable so far. :+1:
